include/dali/core/tensor_shape.h

```diff
@@ -1435,23 +1435,40 @@ auto permute_dims(const TensorListShape<in_ndim> &in, const Permutation &axis_or
 }
 
 /**
- * Returns new shape, where samples are subset of samples in given TensorListShape.
+ * Stores a (strided) range of sample shapes in the output list shape.
  *
- * @param begin Start index, included in the output
- * @param end End index, excluded from the output
+ * @param begin      index of the first sample to include in the subrange
+ * @param end        index one past the last sample to include in the subrange
+ * @param step       stride between consecutive source samples
  */
-template <int ndims>
-TensorListShape<ndims> subshape(const TensorListShape<ndims> &in, int begin, int end,
-                                int step = 1) {
-  assert(step >= 1);
-  assert(end > begin);
-  assert(begin >= 0);
-  auto div_ceil = [](int x, int y) { return 1 + ((x - 1) / y); };
+template <int out_ndim, int ndim>
+void sample_range(TensorListShape<out_ndim> &out, const TensorListShape<ndim> &in,
+                  int begin, int end, int step = 1) {
+  detail::check_compatible_ndim<out_ndim, ndim>();
+  assert(begin >=0 && begin <= in.num_samples());
+  assert(end >= begin && end <= in.num_samples());
```

Comment:
Please add test for corner case `begin=0, end=0` (or `begin == end` in general)

Reply:
Done.